Assets/LeapMotionModules/Interaction/InteractionManager.cs

```diff
@@ -0,0 +1,650 @@
+        if (enableGrasping) {
+          RemapInteractionObjectStateChecks(
+            hands: _interactionHands,
+            stateCheckFunc: (InteractionHand maybeBeganGraspingHand, out IInteractionBehaviour graspedObject) => {
+              return maybeBeganGraspingHand.CheckGraspBegin(out graspedObject);
+            },
+            actionPerInteractionObject: (newlyGraspedObject, beganGraspingIntHands) => {
+              newlyGraspedObject.BeginGrasp(beganGraspingIntHands);
+            });
+        }
+
+        // Sustained Interactions //
+
+        // Check sustaining hover.
+        if (enableHovering) {
+          RemapMultiInteractionObjectStateChecks(
+            hands: _interactionHands,
+            multiObjectStateCheckFunc: (InteractionHand maybeSustainedHoveringHand, out HashSet<IInteractionBehaviour> hoveredObjects) => {
+              return maybeSustainedHoveringHand.CheckHoverStay(out hoveredObjects);
+            },
+            actionPerInteractionObject: (hoveredObject, hoveringIntHands) => {
+              hoveredObject.StayHovered(hoveringIntHands);
+            });
+        }
+
+        // Check sustaining primary hovers.
+        if (enableHovering) {
+          RemapInteractionObjectStateChecks(
+            hands: _interactionHands,
+            stateCheckFunc: (InteractionHand maybeSustainedPrimaryHoveringHand, out IInteractionBehaviour primaryHoveredObject) => {
+              return maybeSustainedPrimaryHoveringHand.CheckPrimaryHoverStay(out primaryHoveredObject);
+            },
+            actionPerInteractionObject: (primaryHoveredObject, primaryHoveringHands) => {
+              primaryHoveredObject.StayPrimaryHovered(primaryHoveringHands);
+            });
+        }
+
+        // Check sustained contact.
+        if (enableContact) {
+          RemapMultiInteractionObjectStateChecks(
+            hands: _interactionHands,
+            multiObjectStateCheckFunc: (InteractionHand maybeSustainedContactingHand, out HashSet<IInteractionBehaviour> contactedObjects) => {
+              return maybeSustainedContactingHand.CheckContactStay(out contactedObjects);
+            },
+            actionPerInteractionObject: (contactedObject, contactingIntHands) => {
+              contactedObject.StayContacted(contactingIntHands);
+            });
+        }
+
+        // Check sustained grasping.
+        if (enableContact) {
+          RemapInteractionObjectStateChecks(
+            hands: _interactionHands,
+            stateCheckFunc: (InteractionHand maybeSustainedGraspingHand, out IInteractionBehaviour graspedObject) => {
+              return maybeSustainedGraspingHand.CheckGraspHold(out graspedObject);
+            },
+            actionPerInteractionObject: (contactedObject, contactingIntHands) => {
+              contactedObject.StayGrasped(contactingIntHands);
+            });
+        }
+
+      }
+    }
+    
+    private delegate V StateChangeCheckFunc<H, T, V>(H hand, out T obj);
```

Comment:
Any reason these delegates are generic?  They are only used once, so you could just put the type declarations right here instead.